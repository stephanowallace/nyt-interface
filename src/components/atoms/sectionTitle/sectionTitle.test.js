import React from 'react';
import { shallow } from 'enzyme';
import SectionTitle from '.';

describe('Section title', () => {
  it('renders default state', () => {
    const component = shallow(<SectionTitle>test</SectionTitle>);
    expect(component).toMatchSnapshot();
  });

  it('renders loader', () => {
    const component = shallow(<SectionTitle loading>test</SectionTitle>);
    expect(component).toMatchSnapshot();
  });

  it('renders child in an h1 tag', () => {
    const component = shallow(<SectionTitle>test</SectionTitle>);
    expect(component.find('h1').text()).toBe('test');
  });

  it('doesnt render if no child', () => {
    const component = shallow(<SectionTitle />);
    expect(component.find('h1')).toHaveLength(0);
  });
});
