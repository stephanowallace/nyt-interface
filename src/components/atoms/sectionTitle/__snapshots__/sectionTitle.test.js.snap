// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section title renders default state 1`] = `
<h1
  className="title "
>
  test
</h1>
`;

exports[`Section title renders loader 1`] = `
<h1
  className="title "
>
  test
</h1>
`;
